src/components/docs-anchor/index.tsx

```diff
@@ -1,60 +0,0 @@
-import Link from 'next/link'
-import classNames from 'classnames'
-import { useCurrentProduct } from 'contexts'
-import InlineLink from 'components/inline-link'
-import s from 'components/inline-link/inline-link.module.css'
-
-const DocsAnchor = ({
-	href,
-	children,
-	...rest
-}: JSX.IntrinsicElements['a']) => {
-	const currentProduct = useCurrentProduct()
-	const passableProps = {
-		...rest,
-		// TODO: we probably want to make use of the InlineLink component in the future instead of just adopting its styles here. Tackle that when we have bandwidth to VQA lots of links.
-		className: classNames(rest.className, s.root),
-	}
-
-	if (!href) {
-		return <a {...passableProps}>{children}</a>
-	}
-
-	// Authors write content as if it only exists for their product,
-	// eg Waypoint content contains links that start with "/docs".
-	// We need to be adjust these links, eg to start with "/waypoint/docs".
-	let adjustedHref = href
-	// TODO: infer this condition dynamically
-	if (
-		currentProduct.basePaths.some((basePath) => href.startsWith(`/${basePath}`))
-	) {
-		adjustedHref = `/${currentProduct.slug}${href}`
-	}
-
-	// Render inline link if it's for sure a string and not a heading permalink
-	if (typeof children === 'string' && children !== '»') {
-		return (
-			<InlineLink href={adjustedHref} textWeight="medium">
-				{children}
-			</InlineLink>
-		)
-	}
-
-	// Render a next/link to avoid a full reload if we know for sure the link is an internal docs link.
-	// TODO: This heuristic can likely be tweaked to be more inclusive of other internal links
-	if (adjustedHref !== href) {
-		return (
-			<Link href={adjustedHref}>
-				<a {...passableProps}>{children}</a>
-			</Link>
-		)
-	}
-
-	return (
-		<a href={adjustedHref} {...passableProps}>
-			{children}
-		</a>
-	)
```

Comment:
This last case was originally intended to handle absolute URLs, I think... though as mentioned in a TODO comment above, it might also catch some internal relative URLs.

We've recently discussed how `next/link` can handle absolute URLs without any problems - so as with other cases in this file, rendering `InlineLink` here seems equivalent (only subtlety being that we'll now be adding `hds-typography` and `hds-font-weight` classes from `InlineLink`). The new `MdxA` renders `InlineLink`, so meets this case as well.

Reply:
🗒️ This case handled at least two things:

1. The heading permalinks injected by `anchor-links`, that have the `»` character as `children`
2. Anchor elements with non-string `children`, such as an anchor element that contains a `code` element

I think 2. is covered fine by `MdxA`. Maybe we want to add a child selector for styling `code` elements in `MdxA`? 💭 

As for 1., we shouldn't render `InlineLink` for injected permalinks that have the `»` character. We should render those as-is, without applying anything to them (other than the CSS that's currently hiding them). Since this would need to be considered for both Learn and Docs content, I think we have two options.

- Update `MdxA` to check `typeof children === 'string' && children === '»'` and render a plain anchor if true. Otherwise, render `InlineLink`.
- Create another component that checks `typeof children === 'string' && children === '»'` and renders a plain anchor if true. Otherwise, renders `MdxA`.

I think we should also consider rendering plain anchor elements for the additional anchors injected by `anchor-links` with empty `href`s. We definitely want to make sure those stay not-visible, so we probably shouldn't be applying any CSS to them like `InlineLink` does.

Reply:
@ashleemboyer as far as I can tell, since we were already passing `className` imported from `InlineLink` here, the only difference here in switching to `MdxA` is:

1. `Link` rather than `<a />` is used
2. Additional `hds-` classes are applied.

I believe the CSS to hide the `»` links is still functional, so I don’t think either of these differences are significant. If that sounds right, I’d love to avoid the additional conditional here, and defer the `»` styling concerns to a different PR (eg, when we implement and show those elements more intentionally - it seems likely this will be through the MdxHeading component and won’t collide with MdxA). Would you be on board with deferring this concern, as long as the `»` links are still hidden?

Reply:
@ashleemboyer Based on discussion in Slack, have gone with `1.` that you suggested. Let me know if that looks alright to you! 🙇